cli/src/commands/branch/forget.rs

```diff
@@ -47,8 +47,13 @@ pub fn cmd_branch_forget(
     let repo = workspace_command.repo().clone();
     let matched_branches = find_forgettable_branches(repo.view(), &args.names)?;
     let mut tx = workspace_command.start_transaction();
-    for (name, _) in &matched_branches {
-        tx.mut_repo().remove_branch(name);
+    for (name, branch_target) in &matched_branches {
+        tx.mut_repo()
+            .set_local_branch_target(name, RefTarget::absent());
+        for (remote_name, _) in &branch_target.remote_refs {
```

Comment:
Thanks for this. I ran into a mess with `jj branch foget --local`; I think this should make it much clearer how things should go.